middleware/app/Models/ProviderUserResource.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class ProviderUserResource extends Model
+{
+    use HasFactory;
+    protected $fillable = [
+        "user_id",
+        "provider_resource_id",
+        "provider_id",
+        "completed",
```

Comment:
Should or could this be an enum to leave room for multiple states? (in_progress, etc.)